ActivityEmbeddingWithPredictiveBack/app/src/main/java/com/google/sample/ae/pb/MainActivity.kt

```diff
@@ -0,0 +1,146 @@
+ * (https://developer.android.com/guide/topics/large-screens/activity-embedding) with a focus on
+ * embedding another app when there is enough space on the screen.
+ *
+ * It will also show how to start another app in multi-window mode from the host app.
+ */
+
+class MainActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContent {
+            ActivityEmbeddingAndPredictiveBackTheme {
+                Surface(
+                    modifier = Modifier.fillMaxSize(),
+                ) {
+                    HostUi(
+                        multiWindow = this::launchSecondActivityInMultiWindow,
+                        activityEmbedding = this::launchSecondActivityEmbedded
+                    )
+                }
+            }
+        }
+    }
+
+    /***
+     * This method launches the embedded app with Activity Embedding API, so that on areas wider
+     * than 600dp two activities can be displayed at the same time. On smaller areas the embedded
+     * activity will just open at full screen.
+     */
+    private fun launchSecondActivityEmbedded() {
+        val intent = Intent().apply {
+            component = ComponentName(
+                "com.google.sample.ae.guestapp",
+                "com.google.sample.ae.guestapp.MainGuestActivity"
+            )
+            addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP)
+        }
+        startActivity(intent)
+    }
+
+    /***
+     * This method launches the target Activity in Multi-Window mode: it will split the screen in
+     * half - either vertically on compact screens, or horizontally on medium and expanded screens -
+     * and display the two activities side by side
+     */
+    private fun launchSecondActivityInMultiWindow() {
+        val intent = Intent().apply {
+            component = ComponentName(
+                "com.google.sample.ae.guestapp",
+                "com.google.sample.ae.guestapp.MainGuestActivity"
+            )
+            addFlags(Intent.FLAG_ACTIVITY_LAUNCH_ADJACENT)
+            addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+            addFlags(Intent.FLAG_ACTIVITY_MULTIPLE_TASK)
+        }
+        startActivity(intent)
+    }
+}
+
+@Composable
+fun HostUi(
+    modifier: Modifier = Modifier,
+    multiWindow: () -> Unit = {},
+    activityEmbedding: () -> Unit = {}
+) {
+
+    Column(
+        modifier = modifier
+            .fillMaxSize()
+            .background(MaterialTheme.colorScheme.secondary),
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        Text(
+            text = "I'm the host app!",
+            color = MaterialTheme.colorScheme.onSecondary,
+            style = MaterialTheme.typography.headlineMedium
+        )
+
+        Spacer(modifier = Modifier.height(24.dp))
+
+        Button(onClick = activityEmbedding) {
+            Text(text = "Launch detail embedded")
```

Comment:
Nit: I think it'd be good if the strings are in `strings.xml` as opposed to hardcoded.